api_factory/generator/templates/$namespace/$name_$version/$service/client.py.j2

```diff
@@ -97,6 +100,38 @@ class {{ service.name }}:
             default_timeout=None,  # FIXME
             client_info=self.client_info,
         )
+        {%- if method.field_headers %}
+
+        # Ensure metadata is a mutable sequence.
+        if metadata is None:
+            metadata = []
+        else:
+            metadata = list(metadata)
+
+        # Add request headers to metadata.
+        try:
+            request_headers = (
+                {%- for field_header in method.field_headers %}
+                ('{{ field_header.header }}', '{{ field_header.field }}', request.{{ field_header.field }}),
+                {%- endfor %}
+            )
+        except AttributeError:
+            # Suppress when attribute missing.
```

Comment:
Is this correct? Seems like this will make _any_ attribute's being missing suppress _all_ of them.

Reply:
Also, since this is protocol buffers, the attribute is likely to be present-but-falsy often; do you want a truthiness check?

Reply:
I couldn't think of a way to check the individual fields without being super repetitive or depending on an external function. Since most of the existing things that use request header params use 1 or at most 2 fields, I figured it wouldn't hurt too much to throw the whole thing away for one missing attribute.

`gapic-generator` has [the same problem](https://github.com/googleapis/gapic-generator/blob/master/src/test/java/com/google/api/codegen/testdata/py/python_library.baseline#L1564). I'm wondering if moving more of this logic to `api_core` is a better solution for both generators.